from Bio import SeqIO
import sys
import random
from tqdm import tqdm

#label matrix of k-mers with taxonmic rank, from class to genus
#parameters: argv[1] = matrix of k-mers, as generated by fasta2matrix script; arg[2] = taxonomy file (e.g. taxonomy.txt")
nome = sys.argv[1].split(".")[0]

outputFileClass = open(nome + "_C.txt", "w")
outputFileOrder = open(nome + "_O.txt", "w")
outputFileFamily = open(nome + "_F.txt", "w")
outputFileGenus = open(nome + "_G.txt", "w")

matrix = list(open(sys.argv[1], "r"))
matrix = matrix[1:]

records = open(sys.argv[2], "r")

record_dict = {}

for seq in records:
    elements = seq.split(" ")
    sequ = elements[0][1:]
    classe = elements[1]
    ordine = elements[2]
    famiglia = elements[3]
    genere = elements[4]

    record_dict[sequ] = dict(classe=classe,
                             ordine=ordine,
                             famiglia=famiglia,
                             genere=genere)

for line in tqdm(matrix):
    sequ2num = line.split(",")[0]
    sequ2 = sequ2num.split("-")[0]
    record = record_dict[sequ2]
    outputFileGenus.write(line.split("\n")[0] + "," + record["genere"])
    outputFileFamily.write(line.split("\n")[0] + "," + record["famiglia"] + "\n")
    outputFileOrder.write(line.split("\n")[0] + "," + record["ordine"] + "\n")
    outputFileClass.write(line.split("\n")[0] + "," + record["classe"] + "\n")
